src/main/scala/net/bmjames/opts/types/Doc.scala

```diff
@@ -0,0 +1,199 @@
+package net.bmjames.opts.types
+
+import scalaz._, Scalaz._
+import scala.annotation.tailrec
+
+//Based on the paper http://homepages.inf.ed.ac.uk/wadler/papers/prettier/prettier.pdf
+
+sealed abstract class Doc
+
+//No text
+case object EmptyDoc extends Doc
+//A line break
+case object LineDoc extends Doc
+//Nest i spackes after any line break in the Doc.
+case class NestDoc private (i: Int, d: Doc) extends Doc {
```

Comment:
I think that might fit my needs better. Initially I only had `TextDoc` with a private constructor, because it can't have `\n` in it.